gui/src/3D/controls/touchButton3D.ts

```diff
@@ -250,30 +249,37 @@ export class TouchButton3D extends Button3D {
             newPushDepth = Math.max(newPushDepth, value);
         });
 
-        if (newPushDepth == ButtonState.Press) {
-            if (previousPushDepth == ButtonState.Hover) {
+        this._firePointerEvents(newPushDepth, previousPushDepth, pointOnButton);
+    }
+
+    protected _firePointerEvents(newButtonState: ButtonState, previousButtonState: ButtonState, pointOnButton: Vector3) {
+        const dummyPointerId = 0;

```

Comment:
If i understand correctly, this ID will not be used outside of this scope, but `0` as ID is the first a pointer will receive. that might confuse developers who relay on pointerId for pointer detection. It aht a problem in this scenario?

Reply:
We're also going to need to eventually be able to distinguish between multiple pointers (near/far, left/right hand). Might be out of scope for this PR, but should we come up with special values to signify those pointers? Or maybe pass an  extra enum value to distinguish between id's for traditional pointers and id's for holographic hand/controller pointers?

Reply:
It hasn't caused any issues yet, though I also don't think anyone has really used this class yet either. I originally left it as 0 since I couldn't figure out how to get the pointer ID from the hand.

Reply:
The pointer ID can be set by you, as long as it is persistent between calls. This is what being done in the pointer-selection feature - https://github.com/BabylonJS/Babylon.js/blob/master/src/XR/features/WebXRControllerPointerSelection.ts#L424 . Each connected controller gets a new ID, and keeps it for as long as it is connected.